mempool/src/core_mempool/transaction_store.rs

```diff
@@ -143,27 +160,6 @@ impl TransactionStore {
         OP_COUNTERS.set("txn.priority_index", self.priority_index.size());
     }
 
-    /// checks if Mempool is full
-    /// If it's full, tries to free some space by evicting transactions from ParkingLot
-    /// We only evict on attempt to insert a transaction that would be ready for broadcast upon insertion
-    fn check_if_full(&mut self, txn: &MempoolTransaction, curr_sequence_number: u64) -> bool {
```

Comment:
would argue the naming isn't great "check_if_full" implies read-only

maybe garbage_collect is a better term?